### ISSUE TITLE
InvalidArgument, Database name "parent=projects%2Fmy-projects%2Fdatabases%2F%28default%29%2Fdocuments" lacks "projects" at index 0.

### ISSUE THREAD

**Issue by clement128** [googleapis/google-cloud-dotnet#11653]
Encountered Grpc.Core.RpcException `InvalidArgument, Database name "parent=projects%2Fmy-projects%2Fdatabases%2F%28default%29%2Fdocuments" lacks "projects" at index 0.` error while connecting to local firestore emulator after upgrade `Google.Cloud.Firestore` to version `3.5.0`, but it was working fine in version `3.5.0-beta01`

#### Environment details

  - OS:  Mac OS X 14.2 osx-arm64
  - .NET version: SDK 8.0.100, Host 8.0.1
  - Package name and version: 
      - Google.Cloud.Firestore = 3.5.0

####  

```f#
#r "nuget: Google.Cloud.Firestore, 3.5.0"

open Google.Api.Gax
open Google.Cloud.Firestore

System.Environment.SetEnvironmentVariable("FIRESTORE_EMULATOR_HOST", "localhost:8123")

let db = 
    FirestoreDbBuilder(
        EmulatorDetection = EmulatorDetection.EmulatorOnly,
        ProjectId = "my-projects"
    ).Build()

let run () =
    task {
      let data = dict [ "Name", "admin" ]
      let! result = db.Collection("Users").Document("admin").CreateAsync data
      return result
    }

run().Wait()

```

```log
System.AggregateException: One or more errors occurred. (Status(StatusCode="InvalidArgument", Detail="Database name "database=projects%2Fmy-projects%2Fdatabases%2F%28default%29" lacks "projects" at index 0."))
 ---> Grpc.Core.RpcException: Status(StatusCode="InvalidArgument", Detail="Database name "database=projects%2Fmy-projects%2Fdatabases%2F%28default%29" lacks "projects" at index 0.")
   at Google.Api.Gax.Grpc.ApiCallRetryExtensions.<>c__DisplayClass0_0`2.<<WithRetry>b__0>d.MoveNext() in /_/Google.Api.Gax.Grpc/ApiCallRetryExtensions.cs:line 39
--- End of stack trace from previous location ---
   at Google.Cloud.Firestore.WriteBatch.CommitAsync(ByteString transactionId, CancellationToken cancellationToken) in /_/apis/Google.Cloud.Firestore/Google.Cloud.Firestore/WriteBatch.cs:line 225
   at Google.Cloud.Firestore.DocumentReference.CreateAsync(Object documentData, CancellationToken cancellationToken) in /_/apis/Google.Cloud.Firestore/Google.Cloud.Firestore/DocumentReference.cs:line 106
```

Thanks!

**Comment by amanda-tarafa** [comment]
I'll try to take a look later today, else tomorrow.

In the meantime, please consider reverting to Google.Cloud.Firestore 3.5.0-beta01. Quickly looking at the changes from 3.5.0-beta01 => 3.5.0 nothing jumps at me as possibly having this effect.

**Comment by jskeet** [comment]
@amanda-tarafa: I suspect this is to do with changing from the legacy resource header to the new one. My guess is that the emulator has the same problem that we found in Listen.

**Comment by amanda-tarafa** [comment]
Oh yes, that makes sense. I'll check internally.

**Comment by jskeet** [comment]
@clement128 Please could you let us know which Firestore emulator you're using? There's one in the Firebase emulator suite, and the more standalone one launched via gcloud.

**Comment by clement128** [comment]
hi @jskeet I am using the Firebase emulator suit, here is the command line:

```
firebase emulators:start --import=./data --export-on-exit
```

firesbase.json
```json
{
  "emulators": {
    "singleProjectMode": true,
    "firestore": {
      "port": 8123
    },
    "ui": {
      "enabled": true
    }
  }
}

```

firebase version
```
firebase -V
13.1.0
```

java version
```
java -version
openjdk version "17.0.6" 2023-01-17 LTS
OpenJDK Runtime Environment Zulu17.40+19-CA (build 17.0.6+10-LTS)
OpenJDK 64-Bit Server VM Zulu17.40+19-CA (build 17.0.6+10-LTS, mixed mode, sharing)
```

**Comment by jskeet** [comment]
Good to know, thanks for the swift response.

**Comment by jskeet** [comment]
As an FYI, we're going to discuss whether to undo some of the changes (that were originally made to fix a different issue) and release a new version. (There's discussion internally about fixing the emulator.)

**Comment by jskeet** [comment]
This will be released in version 3.5.1 (in progress).
